lib/resource-mapper.js

```diff
@@ -68,6 +68,12 @@ class ResourceMapper {
     if (path.startsWith(this._rootPath)) {
       path = path.substring(this._rootPath.length)
     }
+    if (this._includeHost) {
+      if (!path.startsWith(`/${hostname}`)) {
+        throw new Error(`Path must start with hostname (/${hostname})`)
+      }
+      path = path.substring(`/${hostname}`.length)
```

Comment:
No need to create a new string just for length; let' take `hostname.length + 1`.